includes/libraries/tracks/class-tracks-client.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * Send Tracks events on behalf of a user.
+ *
+ * @class   Tracks_Client
+ * @package WooCommerce/Classes
+ *
+ *
+ * Example Usage:
+```php
+	require( dirname(__FILE__).'path/to/tracks/class-tracks-client.php' );
+
+	$result = Tracks_Client::record_event( array(
+		'_en'        => $event_name,       // required
+		'_ui'        => $user_id,          // required unless _ul is provided
+		'_ul'        => $user_login,       // required unless _ui is provided
+
+		// Optional, but recommended
+		'_ts'        => $ts_in_ms,         // Default: now
+		'_via_ip'    => $client_ip,        // we use it for geo, etc.
+
+		// Possibly useful to set some context for the event
+		'_via_ua'    => $client_user_agent,
+		'_via_url'   => $client_url,
+		'_via_ref'   => $client_referrer,
+	) );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+
+require_once dirname( __FILE__ ) . '/class-tracks-event.php';
+
+/**
+ * Class Tracks_Client.
+ */
+class Tracks_Client {
```

Comment:
As you said, this library is inspirited by Jetpack, but seems modified for WooCommerce at least, since there's options only available on WooCommerce and also filters... So this class name seems generic, probably will cause problems, we probably could move this to another folder and stop look at it as a external library.